homeassistant/components/elmax/config_flow.py

```diff
@@ -60,20 +120,205 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
 
     VERSION = 1
     _client: Elmax
-    _username: str
-    _password: str
+    _selected_mode: str
+    _panel_pin: str
+    _panel_id: str
+
+    # Direct API variables
+    _panel_direct_use_ssl: bool
+    _panel_direct_hostname: str
+    _panel_direct_port: int
+    _panel_direct_follow_mdns: bool
+    _panel_direct_ssl_cert: str | None
+    _panel_direct_http_port: int
+    _panel_direct_https_port: int
+
+    # Cloud API variables
+    _cloud_username: str
+    _cloud_password: str
+    _reauth_cloud_username: str | None
+    _reauth_cloud_panelid: str | None
+
+    # Panel selection variables
     _panels_schema: vol.Schema
     _panel_names: dict
     _entry: config_entries.ConfigEntry | None
+    _zeroconf_panel_api_uri: str
 
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
-        """Handle a flow initialized by the user."""
-        # When invokes without parameters, show the login form.
+        """Handle the flow initiated by the user."""
+        return await self.async_step_choose_mode(user_input=user_input)
+
+    async def async_step_choose_mode(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle local vs cloud mode selection step."""
+        errors: dict[str, Any] = {}
         if user_input is None:
-            return self.async_show_form(step_id="user", data_schema=LOGIN_FORM_SCHEMA)
+            return self.async_show_form(
+                step_id="choose_mode", data_schema=CHOOSE_MODE_SCHEMA, errors=errors
+            )
+
+        self._selected_mode = user_input[CONF_ELMAX_MODE]
+        if self._selected_mode == CONF_ELMAX_MODE_CLOUD:
+            return self.async_show_form(
+                step_id="cloud_setup", data_schema=LOGIN_FORM_SCHEMA, errors=errors
+            )
+        # Assume mode direct.
+        return self.async_show_form(
+            step_id="direct_setup", data_schema=DIRECT_SETUP_SCHEMA, errors=errors
```

Comment:
On second thought, you're never setting an error, so why bother storing it and passing it to the forms?